tiledb/cf/core.py

```diff
@@ -353,28 +328,54 @@ def create_virtual(
                 decrypt arrays.
             ctx: If not ``None``, TileDB context wrapper for a TileDB storage manager.
         """
-        cls.create(uri, array_schemas, group_metadata, key, ctx, is_virtual=True)  # pragma: no cover
+        cls.create(uri, group_schema, key, ctx, is_virtual=True)  # pragma: no cover
 
     def __init__(
         self,
         uri: str,
+        mode: str = "r",
         key: Optional[Union[Dict[str, str], str]] = None,
         timestamp: Optional[int] = None,
+        array: Optional[str] = None,
         attr: Optional[str] = None,
         ctx: Optional[tiledb.Ctx] = None,
     ):
         """Constructs a new :class:`Group`."""
-        self.uri = uri
-        self.key = key
-        self.timestamp = timestamp
-        self.ctx = ctx
-        self.open_arrays = dict()
-
-        self._attr = attr
+        self._group_schema = GroupSchema.load(uri, ctx, key)
+        self._array_uris = {
+            array_name: _get_array_uri(uri, array_name, False)
+            for array_name in self._group_schema.keys()
+        }
+        self._metadata_array = (
+            None
+            if self._group_schema.metadata_schema is None
+            else tiledb.open(
+                uri=_get_metadata_array_uri(uri, False),
+                mode=mode,
+                key=_get_array_key(key, METADATA_ARRAY_NAME),
+                timestamp=timestamp,
+                ctx=ctx,
+            )
+        )
+        self._mode = mode
+        self._key = key
+        self._timestamp = timestamp
+        self._ctx = ctx
+        self._opened_arrays: List[tiledb.Array] = []
```

Comment:
If we use a Dict[array_name, array] here instead that would allow closing a specific array

Reply:
This requires extra logic the `open_array`. Since you can already close the array directly from the object `open_array` returns, I followed the YAGNI principle and changed it to a list.

I don't have a strong opinion on this, and if you decide that you would like to add the `close_array` method to the class, make sure you use (array_name, attr_name) for the key so the same array can be opened multiple times by different attributes.

Reply:
Got it